mcfly/modelgen.py

```diff
@@ -115,630 +60,64 @@ def generate_models(x_shape,
     
+    # Set default hyperparameter ranges
+    defaults = {'low_lr': 1,
+                'high_lr': 4,
+                'low_reg': 1,
+                'high_reg': 4}
+
+    default_models = {'CNN': Model_CNN,
+                      'DeepConvLSTM': Model_ConvLSTM,
+                      'ResNet': Model_ResNet,
+                      'InceptionTime': Model_InceptionTime}
+
+    # Replace default hyperparameter ranges with input
+    for key, value in hyperparameter_ranges.items():
+        if key in defaults:
+            print("The value of {} is set from {} (default) to {}".format(key, defaults[key], value))
+            defaults[key] = value
+
+    # Add missing parameters from default
+    for key, value in defaults.items():
+        if key not in hyperparameter_ranges:
+            hyperparameter_ranges[key] = value
+
     model_types_selected = []
-    model_types = settings['model_types']
 
     for i in range(int(np.ceil(number_of_models/len(model_types)))):
         np.random.shuffle(model_types)
         model_types_selected.extend(model_types)
 
-    # Create list of Keras models
+    # Create list of Keras models and their hyperparameters
     # -------------------------------------------------------------------------
     models = []
-    for i in range(0, number_of_models):
+    for i in range(number_of_models):
         current_model_type = model_types_selected[i]
-        generate_model = None
-        if current_model_type == 'CNN':
-            generate_model = generate_CNN_model  # generate_model is a function
-            hyperparameters = generate_CNN_hyperparameter_set(settings)
-        if current_model_type == 'DeepConvLSTM':
-            generate_model = generate_DeepConvLSTM_model
-            hyperparameters = generate_DeepConvLSTM_hyperparameter_set(settings)
-        if current_model_type == 'ResNet':
-            generate_model = generate_resnet_model
-            hyperparameters = generate_resnet_hyperparameter_set(settings)
-        if current_model_type == 'InceptionTime':
-            generate_model = generate_InceptionTime_model
-            hyperparameters = generate_InceptionTime_hyperparameter_set(settings)
-
-        models.append(
-            (generate_model(x_shape, 
-                            number_of_classes, 
-                            metrics=settings['metrics'], **hyperparameters),
-             hyperparameters, current_model_type))
+        if current_model_type in default_models:
+            model_type = default_models[current_model_type](x_shape, number_of_classes,
+                                                            metrics, **hyperparameter_ranges)
+            hyperparameters = model_type.generate_hyperparameters()
+            model = model_type.create_model(**hyperparameters)
+            model_name = current_model_type
+        elif isinstance(current_model_type, str):
+            raise NameError("Unknown model name given.")
```

Comment:
```suggestion
            raise NameError("Unknown model name, '{}'.".format(current_model_type)
```

Reply:
Also, can this error be raised in its own loop over the modeltypes?
```
for model_type in model_types:
    if type(model_type) is str and mode_type not in default_models:
        raise NameError(...)
```
For example directly after setting default parameters.

Reply:
Done.